source/mir/random/flex/internal/area.d

```diff
@@ -0,0 +1,668 @@
+            squeeze = mixin(t_r);
+            hat = mixin(t_l);
+            break;
+        case T1b:
+            squeeze = mixin(t_l);
+            hat = mixin(t_r);
+            break;
+        case T2a:
+            squeeze = mixin(sec);
+            hat = mixin(t_l);
+            break;
+        case T2b:
+            squeeze = mixin(sec);
+            hat = mixin(t_r);
+            break;
+        case T3a:
+            squeeze = mixin(t_r);
+            hat = mixin(sec);
+            break;
+        case T3b:
+            squeeze = mixin(t_l);
+            hat = mixin(sec);
+            break;
+        case T4a:
+            if (iv.lx == -S.infinity)
+            {
+                squeeze = squeeze.init;
+                hat = mixin(t_r);
+                break;
+            }
+            if (iv.rx == +S.infinity)
+            {
+                squeeze = squeeze.init;
+                hat = mixin(t_l);
+                break;
+            }
+            squeeze = mixin(sec);
+            hat = iv.ltx > iv.rtx ? mixin(t_l) : mixin(t_r);
+            break;
+        case T4b:
+            squeeze = iv.ltx < iv.rtx ? mixin(t_l) : mixin(t_r);
+            hat = mixin(sec);
+            break;
+        default:
+            squeeze = linearFun!S(0, 0, 0);
+            hat = linearFun!S(0, 0, 0);
+    }
+}
+
+// TODO: add more tests
+unittest
+{
+    import std.meta : AliasSeq;
+    import mir.random.flex.internal.types: determineType;
+    foreach (S; AliasSeq!(float, double, real))
+    {
+        const f0 = (S x) => x * x;
+        const f1 = (S x) => 2 * x;
+        const f2 = (S x) => 2.0;
+        auto c = 42; // not required for this test
+        auto dhs = (S l, S r) {
+            auto iv = Interval!S(l, r, c, f0(l), f1(l), f2(l),
+                                          f0(r), f1(r), f2(r));
+            determineSqueezeAndHat(iv);
+            return iv;
+        };
+
+        // test left side
+        auto hs1 = dhs(-1, 1);
+        assert(hs1.hat.slope == 0);
+        assert(hs1.hat.intercept == 1);
+        assert(hs1.squeeze.slope == 2);
+        assert(hs1.squeeze.intercept == -1);
+
+        // test right side
+        auto hs2 = dhs(1, 3);
+        assert(hs2.hat.slope == 4);
+        assert(hs2.hat.intercept == -3);
+        assert(hs2.squeeze.slope == 2);
+        assert(hs2.squeeze.intercept == -1);
+    }
+}
+
+/**
+Flex-specific constants cutoffs for numeric errors.
+*/
+template constants(S)
+{
+    import mir.internal.math: sqrt;
+    enum S smallLog = sqrt(sqrt(S.epsilon * 5));
+    enum S smallExp = sqrt(sqrt(S.epsilon * 120));
+}
+
+alias hatArea(S) = area!(true, S);
+alias squeezeArea(S) = area!(false, S);
+
+/**
+Computes the area below either the hat or squeeze function
+in-between a interval `iv`.
+Based on table 1 and general equation (3) from Botts et al. (2013).
+
+    (F_T(sh(r))- F_T(sh(l))) / sh.slope
+
+Params:
+    isHat = whether to calculate the hat or squeeze area
+    iv = Interval to use
```

Comment:
> Hmmmm. To me it just looks a bit weird to have one undocumented parameter in the middle, even though (as you say) it's pretty obvious what it is.

Well my point was that we never document `S` in all other methods, so at least we are consistent. 

>  But anyway, not a big deal.

I don't mind adding it, @9il what do you say?